lib/rules/no-extra-parens.js

```diff
@@ -860,6 +872,22 @@ module.exports = {
                 if (isReturnAssignException(node)) {
                     return;
                 }
+
+                const availableType = new Set(["BinaryExpression", "LogicalExpression"]);
+
+                if (isCondTernaryException(node)) {
+                    if (!availableType.has(node.test.type) && isParenthesised(node.test)) {
+                        report(node.test);
+                    }
+                    if (!availableType.has(node.consequent.type) && isParenthesised(node.consequent)) {
+                        report(node.consequent);
+                    }
+                    if (!availableType.has(node.alternate.type) && isParenthesised(node.alternate)) {
+                        report(node.alternate);
+                    }
+                    return;
+                }
```

Comment:
These would be false positives:

```js
/* eslint no-extra-parens: ["error", "all", { conditionalTernary: false }] */

(a, b) ? (c, d) : (e, f); // false positives

```

We should just add the new logic into the existing conditionals below, for example:

```diff
-if (hasExcessParensWithPrecedence(node.alternate, PRECEDENCE_OF_ASSIGNMENT_EXPR)) {
+if (
+    !(EXCEPT_COND_TERNARY && availableType.has(node.alternate.type)) &&
+    hasExcessParensWithPrecedence(node.alternate, PRECEDENCE_OF_ASSIGNMENT_EXPR)
+) {
    report(node.alternate);
}
```

Reply:
It seems that this problem can be solved by adding `SequenceExpression` to `availableType`.

Reply:
SequenceExpression is just one example, the existing logic handles more. For example, the following parentheses are required because of operator precedence and thus shouldn't be reported:

```js
/* eslint no-extra-parens: ["error", "all", { "conditionalTernary": false } ] */

(a = b) ? c : d; // false positive, removing parentheses will change semantics to a = (b ? c : d)

```

I think we shouldn't be really adding another branch for the new option, it seems best to just add new checks into the existing conditionals, as suggested in my previous comment.